### PR TITLE
Refactor GetRandomSignedHeaders and GetValidatorSet Helpers

### PR DESCRIPTION
**Pull request by AryanGodara**
Cleaning up helper methods to adhere to a common pattern of:
```golang
GetXYZ()
GetXYZCustom()
```

Previously, we were heading down a not very DRY path of GetXYZWithABC where ABC was just some config change.
This PR DRYs up these helper methods so that only the Custom methods need to be extended in the future, and the majority of tests can continue to use the basic Get method without 
issue.

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

## Summary by CodeRabbit

- **Refactor**
	- Enhanced the flexibility and customization in block and validator set generation for testing purposes.
- **Tests**
	- Improved test configurations for more accurate simulation of various scenarios.
- **Chores**
	- Renamed several core functions and structs to better reflect their purpose and usage.

### CODE REVIEW

**Comment by AryanGodara** [comment]
@MSevey Ready for review :D

File: types/utils.go [hunk 1]
```diff
@@ -21,58 +22,85 @@ var (
...
-			{PubKey: pubKey, Address: pubKey.Address(), VotingPower: 1},
+			{PubKey: pubKey, Address: pubKey.Address(), VotingPower: config.VotingPower},
 		},
 	)
 	return valset
 }
 
-// GetRandomBlock returns a block with random data
+// BlockContext carries all necessary state for block generation, including configurations and outputs.
+type BlockConfig struct {
```
Comment (tzdybal) [inline types/utils.go:56]: The name of this struct is misleading - even later in the code it's argument of this type is called `ctx` in some cases.

File: types/utils.go [hunk 2]
```diff
@@ -21,58 +22,85 @@ var (
...
+	Height       uint64
+	NTxs         int
+	PrivKey      ed25519.PrivKey // Input and Output option
+	ProposerAddr []byte          // Input option
+	Block        *Block          // Output: the generated block.
```
Comment (tzdybal) [inline types/utils.go:61]: I really don't see why one structure is used to both input and output. The typical convention in Go is just to return value.
The code is not simplified. What is the benefit?

**Review by tzdybal — CHANGES_REQUESTED** [review]
Thanks for the submission. I left some comments.

I really don't get where you're going with this refactoring - it's 'counter-idiomatic' to me :wink:

File: types/utils.go (hunk 2 above)
Comment (AryanGodara) [reply types/utils.go:61]: I believe I should change this to return the Block and the private key.
I would return the block, but sometimes, the private key is also expected in the return value.
But you are right, having a common struct is not a clean solution 😞
I'll push a better(hopefully) solution asap!

**Comment by AryanGodara** [comment]
@tzdybal I've updated the code. Got stuck here for a while because of tests failing. But I'm not able to figure out if I've made a breaking change with these modifications.
Can you please help me out with this. If anything looks broken, and you point it out; I'll try to fix it asap.

**Comment by AryanGodara** [comment]
@MSevey @tzdybal 
Since this one is almost ready (🤞🏼); I wanted your opinion on the previous change of changing NodeType from string to an enum; it was one of the smaller changes in the original big PR, and I'm not sure if it's big enough to open a separate PR for. Wanted your opinion on this one :D

```go
type NodeType int

const (
	Full NodeType = iota
	Light
)
```

**Comment by MSevey** [comment]
> @MSevey @tzdybal Since this one is almost ready (🤞🏼); I wanted your opinion on the previous change of changing NodeType from string to an enum; it was one of the smaller changes in the original big PR, and I'm not sure if it's big enough to open a separate PR for. Wanted your opinion on this one :D
> 
> ```go
> type NodeType int
> 
> const (
> 	Full NodeType = iota
> 	Light
> )
> ```

No change is too small :-) 

Smaller PRs are great. Easier to review.

**Review by MSevey — APPROVED** [review]
lgtm.  Since the PR description I think got lost with the break out of smaller PRs, the origin of this PR I believe was cleaning up these helper methods to adhere to a common pattern of:
```go
GetXYZ()
GetXYZCustom()
```
Previously, we were heading down a not very DRY path of `GetXYZWithABC` where `ABC` was just some config change. 
This PR DRYs up these helper methods so that only the `Custom` methods need to be extended in the future and the majority of tests can continue to use the basic Get method without issue.

**Comment by AryanGodara** [comment]
> lgtm. Since the PR description I think got lost with the break out of smaller PRs, the origin of this PR I believe was cleaning up these helper methods to adhere to a common pattern of:
> 
> ```go
> GetXYZ()
> GetXYZCustom()
> ```
> 
> Previously, we were heading down a not very DRY path of `GetXYZWithABC` where `ABC` was just some config change. This PR DRYs up these helper methods so that only the `Custom` methods need to be extended in the future and the majority of tests can continue to use the basic Get method without issue.

Thanks for this summary @MSevey :D
I should've written this in the description myself though 😅 . I'll fix this asap!
***

**Review by Manav-Aggarwal — APPROVED** [review]

**Review by Manav-Aggarwal — APPROVED** [review]

**Review by tzdybal — APPROVED** [review]
LGTM

**Pull request merged**